src/services/LogFileManagerProxy.ts

```diff
@@ -0,0 +1,84 @@
+import {Nullable} from "../typings/common";
+import {DecoderOptions} from "../typings/decoders";
+import {LogLevelFilter} from "../typings/logs";
+import {QueryResults} from "../typings/query";
+import {
+    CursorType,
+    FileSrcType,
+    LogFileInfo,
+    PageData,
+} from "../typings/worker";
+import LogFileManager from "./LogFileManager";
+
+
+class LogFileManagerProxy {
+    logFileManager: Nullable<LogFileManager> = null;
+
+    async loadFile (
+        {decoderOptions, fileSrc, pageSize, cursor, isPrettified}: {
+            decoderOptions: DecoderOptions;
+            fileSrc: FileSrcType;
+            pageSize: number;
+            cursor: CursorType;
+            isPrettified: boolean;
+        },
+        onExportChunk: (logs: string) => void,
+        onQueryResults: (queryProgress: number, queryResults: QueryResults) => void,
+    ): Promise<{fileInfo: LogFileInfo; pageData: PageData}> {
+        const logFileManager = await LogFileManager.create({
+            decoderOptions: decoderOptions,
+            fileSrc: fileSrc,
+            onExportChunk: onExportChunk,
+            onQueryResults: onQueryResults,
+            pageSize: pageSize,
+        });
+
+        this.logFileManager = logFileManager;
+
+        return {
+            fileInfo: {
```

Comment:
i might have posted this comment somewhere else:

let's decouple the page loading behaviour from `loadFile` and return `fileInfo` only. e.g.,
```
        return {
            fileName: logFileManager.fileName,
            isStructuredLog: logFileManager.isStructuredLog,
            numEvents: logFileManager.numEvents,
            onDiskFileSizeInBytes: logFileManager.onDiskFileSizeInBytes,
        };
```